src/software/world/ball_model/linear_ball_model.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include <optional>
+
+#include "software/world/ball_model/ball_model.h"
+
+/**
+ * LinearBallModel is a BallModel that makes the following assumptions about the ball:
+ * 1. no initial spin
+ * 2. slides with a constant sliding friction above a certain speed threshold
+ * 3. rolls with a constant rolling friction below a certain speed threshold
+ */
```

Comment:
These assumptions result in a non-linear model, perhaps `SlideThenRollBallModel` or `SlidingAndRollingBallModel` might be a better name for this class? Or even better, just forget about sliding and rolling, this could just be `TwoFrictionStageBallModel` since we really aren't even doing anything specific to sliding or rolling, we're just applying two different friction coeff's depending on the speed.

Reply:
In order to not block this PR, it might be a good idea to just replace the implementation of this class with a linear, single friction coeff model, then add this more complex model in a future PR?

Reply:
This 2-stage friction model might be tricky to tune.

I agree that maybe we should just add a single friction model then create a real ball model using a free body diagram